metron-platform/metron-test-utilities/src/main/java/org/apache/metron/test/mock/MockHTable.java

```diff
@@ -70,7 +93,7 @@ public static void clear() {
   private final String tableName;
   private final List<String> columnFamilies = new ArrayList<>();
   private HColumnDescriptor[] descriptors;
-
+  private List<Put> putLog;
```

Comment:
Why not do ```private List<Put> putLog = Collections.synchronizedList(new ArrayList<>);``` here?  It would eliminate the manual synchronization in `getPutLog()` and `addToPutLog()` and I doubt there's significant perf impact either way.  Correct me if I'm wrong on that.

To elaborate, `add()` is directly taken care of with synchronized list.  The only other concern is the `getPutLog()`, and Guava's docs state `ImmutableList.copyOf()` is fine to use on synchronized collections.  Looking into their code, it uses of `toArray()` (which is synchronized) instead of iteration to copy the List.